framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -234,12 +235,48 @@ protected void finish(QualifierHierarchy qualHierarchy,
     @SideEffectFree
     @Override
     public String toString() {
-        // TODO: it would be easier to debug if the graph and map were sorted by the key.
-        // Simply creating a TreeMap here doesn't work, because AnnotationMirrors are not comparable.
-        return "Supertypes Graph: " + supertypesGraph.toString() +
-                "\nSupertypes Map: " + String.valueOf(supertypesMap) +
-                "\nTops: " + tops +
-                "\nBottoms: " + bottoms;
+        StringBuilder sb = new StringBuilder();
+        sb.append("Supertypes Graph: ");
+
+        for (Entry<AnnotationMirror, Set<AnnotationMirror>> qual : supertypesGraph.entrySet()) {
+            sb.append("\n\t");
+            sb.append(qual.getKey());
+            sb.append(" = ");
+            sb.append(qual.getValue());
+        }
+
+        sb.append("\nSupertypes Map: ");
+
+        for (Entry<AnnotationMirror, Set<AnnotationMirror>> qual : supertypesMap.entrySet()) {
+            sb.append("\n\t");
+            sb.append(qual.getKey());
+            sb.append(" = [");
+
+            Set<AnnotationMirror> supertypes = qual.getValue();
+
+            // if there's only 1 supertype for this qual, then directly display that in the same row
+            if (supertypes.size() == 1) {
+                sb.append(supertypes.iterator().next());
+            }
+            // otherwise, display each supertype in its own row
```

Comment:
Following on the above comment, I would put this comment in the else block rather than outside it.